src/main/java/seedu/address/logic/commands/PayCommand.java

```diff
@@ -0,0 +1,145 @@
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Updates the payment made by the student identified "
+            + "by the index number used in the displayed Student list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_HOUR + "HOURS PAID\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_HOUR + "3 ";
+
+    public static final String MESSAGE_EDIT_STUDENT_SUCCESS = "%1$s paid $%2$.2f";
+
+    private final Index index;
+    private final double hour;
+
+    /**
+     * @param index of the student in the filtered student list to edit
+     * @param hour number of hours the student paid
+     */
+    public PayCommand(Index index, double hour) {
+        requireNonNull(index);
+
+        this.index = index;
+        this.hour = hour;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Student> lastShownList = model.getFilteredStudentList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+        }
+
+        Student studentToEdit = lastShownList.get(index.getZeroBased());
+        double amountPaid = calculatePaid(studentToEdit, hour);
+        Student editedStudent = createEditedStudent(studentToEdit, amountPaid);
+        String name = editedStudent.getName().toString();
+
+        model.setStudent(studentToEdit, editedStudent);
+        model.updateFilteredStudentList(PREDICATE_SHOW_ALL_STUDENTS);
+        return new CommandResult(String.format(MESSAGE_EDIT_STUDENT_SUCCESS, name, amountPaid));
+
+
+    }
+
+    /**
+     * Creates and returns a {@code Student} with the paid attribute of {@code studentToEdit}
+     * updated with {@code amountPaid}.
+     */
+    private static Student createEditedStudent(Student studentToEdit, double amountPaid) {
+        assert studentToEdit != null;
+
+        Name updatedName = studentToEdit.getName();
+        Phone updatedPhone = studentToEdit.getPhone();
+        Email updatedEmail = studentToEdit.getEmail();
+        Address updatedAddress = studentToEdit.getAddress();
+        Schedule updatedSchedule = studentToEdit.getSchedule();
+        Subject updatedSubject = studentToEdit.getSubject();
+        Rate updatedRate = studentToEdit.getRate();
+        PaidAmount updatedPaid = updatePaidAmount(studentToEdit, amountPaid);
+        OwedAmount updatedOwedAmount = studentToEdit.getOwedAmount();
+
+        return new Student(
+                updatedName, updatedPhone, updatedEmail, updatedAddress, updatedSchedule, updatedSubject, updatedRate,
+                updatedPaid, updatedOwedAmount
+        );
+    }
+
+    /**
+     * Updates the paid amount of the student based on the hours the student paid.
+     */
+    private static PaidAmount updatePaidAmount(Student student, double amountPaid) {
```

Comment:
good to assert here

Reply:
Thanks for the suggestion, Ive added a assert statement